exp/api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -141,77 +141,41 @@ func (m *AzureManagedControlPlane) ValidateUpdate(oldRaw runtime.Object, client
 
 	if old.Spec.DNSServiceIP != nil {
 		// Prevent DNSServiceIP modification if it was already set to some value
-		if m.Spec.DNSServiceIP == nil {
-			// unsetting the field is not allowed
+		if !reflect.DeepEqual(m.Spec.DNSServiceIP, old.Spec.DNSServiceIP) {
```

Comment:
These `reflect.DeepEqual` changes are in response to linter complaining about this func's cyclomatic complexity with the addition of more validations. Next step might be to refactor this more, but this works for now.

Reply:
We should really refactor this to not do all the validation in one big function (maybe add a `ValidateImmutableFields`?), could you please open an issue so we don't forget? Could be a good first issue maybe

we're losing some error differentiation with this change so it's not fully equivalent, are we okay with that?

Reply:
Is something like this an improvement (imagine this applied across all webhook validations that check for immutability):

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2718

Reply:
I'll finish #2718 to completion (audit all webhooks) after this PR lands (I think this PR is ready for final review).

And then after #2718 I'll file an issue that advocates this surface area is a potential good opportunity for introducing go generics into capz codebase.

Reply:
> I'll finish https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2718 to completion (audit all webhooks) after this PR lands (I think this PR is ready for final review)

In that case I would advocate that we temporarily fix the go cyclomatic linter some other way (maybe extract some stuff to a function) so we don't temporarily regress the error messaging

Reply:
How about we just land #2718 first? (this PR is not highest priority)

Reply:
that works too 👍

Reply:
Prereq work is done, this PR is ready for another review round